### ISSUE TITLE
4.5.0 breaks uri's with username/password

### ISSUE THREAD

**Issue by kirbycool** [redis/redis-rb#1038]
Hi,

I haven't had time to fully debug this, but noticed that uri's with username/password seem broken in the 4.5.0 version.

Steps I ran to repro
```
> Redis.new(uri: 'redis://someuser:somepassword@a.redis.server.example.com:1234').info
Traceback (most recent call last):
        1: from (irb):20
Redis::CommandError (WRONGPASS invalid username-password pair)
```

I can try to provide more info if needed. Thanks!

**Comment by tadejm** [comment]
Seeing the same after upgrading to 4.5.0.

**Comment by kirbycool** [comment]
@fatkodima Sorry, `:uri` was a type. I was using the correct `:url` option. I've updated the ticket description.

The redis server is running 6.0.9.

**Comment by allisonphillips** [comment]
I'm wondering if this change could have caused it: https://github.com/redis/redis-rb/commit/17ba0947a253f8fa56d6b3e430fc1a4813bef9a2

Prior to that change, a WRONGPASS CommandError would have been caught and auth would be attempted again as `AUTH password` (and prior to adding ACL support for versions < 6, auth was attempted as `AUTH password`). If the server is using the default user but an incorrect username is being passed to auth, it seems like that commit could prevent auth from succeeding and result in this error.

**Comment by casperisfine** [comment]
Yes, it's likely as it's the only auth related change. It's just weird I can't repro. But it seems that `6.0.16` accepts two arguments to the AUTH command, maybe `6.0.3` doesn't? I'll try to see if I can compile that exact version.

**Comment by casperisfine** [comment]
Ah nevermind, I forgot to recreate the user. Still can't repro even with 6.0.3

**Comment by olleolleolle** [comment]
In My password, I had numbers and alphanumeric

**Comment by casperisfine** [comment]
PR is here: https://github.com/redis/redis-rb/pull/1041

I have to go away from my keyboard for a bit, but I'll cut a `4.5.1` as soon as I can assuming CI is green (couple hours max).
